easybuild/easyblocks/e/elpa.py

```diff
@@ -0,0 +1,211 @@
+            'with_shared': [True, "Enable building of shared ELPA libraries", CUSTOM],
+            'with_single': [True, "Enable building of single precision ELPA functions", CUSTOM],
+            'with_generic_kernel': [True, "Enable building of ELPA generic kernels", CUSTOM],
+        }
+
+        for flag in ELPA_CPU_FEATURE_FLAGS:
+            if flag == 'sse4_2':
+                conf_opt = ['sse', 'sse-assembly']
+            elif flag == 'avx512f':
+                conf_opt = ['avx512']
+            else:
+                conf_opt = [flag]
+
+            for opt in conf_opt:
+                help_msg = "Configure with --enable-%s (if None, auto-detect support for %s)" % (opt, flag.upper())
+                extra_vars['use_%s' % flag] = [None, help_msg, CUSTOM]
+
+        return ConfigureMake.extra_options(extra_vars)
+
+    def __init__(self, *args, **kwargs):
+        """Initialisation of custom class variables for ELPA."""
+        super(EB_ELPA, self).__init__(*args, **kwargs)
+
+        for flag in ELPA_CPU_FEATURE_FLAGS:
+            # fail-safe: make sure we're not overwriting an existing attribute (could lead to weird bugs if we do)
+            if hasattr(self, flag):
+                raise EasyBuildError("EasyBlock attribute '%s' already exists")
+            setattr(self, flag, self.cfg['use_%s' % flag])
+
+        # auto-detect CPU features that can be used and are not enabled/disabled explicitly,
+        # but only if --optarch=GENERIC is not being used
+        if self.cfg['auto_detect_cpu_features']:
+
+            # if --optarch=GENERIC is used, we will not use no CPU feature
+            if build_option('optarch') == OPTARCH_GENERIC:
+                cpu_features = []
+            else:
+                cpu_features = ELPA_CPU_FEATURE_FLAGS
+            self.log.info("CPU features considered for auto-detection: %s", cpu_features)
+
+            # get list of available CPU features, so we can check which ones to retain
+            avail_cpu_features = get_cpu_features()
+
+            # on macOS, AVX is indicated with 'avx1.0' rather than 'avx'
+            if 'avx1.0' in avail_cpu_features:
+                avail_cpu_features.append('avx')
+
+            self.log.info("List of available CPU features: %s", avail_cpu_features)
+
+            for flag in cpu_features:
+                # only enable use of a particular CPU feature if it's still undecided (i.e. None)
+                if getattr(self, flag) is None and flag in avail_cpu_features:
+                    self.log.info("Enabling use of %s (should be supported based on CPU features)", flag.upper())
+                    setattr(self, flag, True)
+
+    def run_all_steps(self, *args, **kwargs):
+        """
+        Put configure options in place for different builds (with and without openmp).
+        """
+
+        # make all builds verbose
+        self.cfg.update('preconfigopts', 'autoreconf &&')
```

Comment:
@micaeljtoliveira How is this related to making the build verbose exactly?

Also, should we try and avoid picking up `autoreconf` from the OS here (i.e. requiring that `Autotools` is listed as a build dep)?

Reply:
@boegel Ups, copy-paste strikes again. I'll fix it.

Reply:
@boegel This prompted me to look if the call to `autoreconf` was really necessary. After a couple of tests, it seems not. Maybe it was at some point in some of the older easyconfigs, but AFAICS that's not the case anymore.